src/dist/replication/lib/replica_stub.cpp

```diff
@@ -853,20 +894,14 @@ void replica_stub::on_node_query_reply_scatter2(replica_stub_ptr this_, gpid gpi
     replica_ptr replica = get_replica(gpid);
     if (replica != nullptr && replica->status() != partition_status::PS_POTENTIAL_SECONDARY)
     {
-        if ((replica->status() == partition_status::PS_INACTIVE ||
-            replica->status() == partition_status::PS_SECONDARY)
-            && now_ms() - replica->last_config_change_time_milliseconds()
-            < _options.gc_memory_replica_interval_ms)
+        if (replica->status() == partition_status::PS_INACTIVE
+            && now_ms() - replica->create_time_milliseconds() < _options.gc_memory_replica_interval_ms)
```

Comment:
this will easily exclude many possibilities, better to use last_config_change_time_milliseconds